dev-tools/build_release.py

```diff
@@ -358,63 +351,44 @@ def find_release_version(src_branch):
-  artifacts.append(os.path.join(path, 'elasticsearch-%s.noarch.rpm' % (release)))
+def artifact_names(release):
+  artifacts = []
+  artifacts.append(os.path.join('distribution/zip/target/releases', 'elasticsearch-%s.zip' % (release)))
+  artifacts.append(os.path.join('distribution/tar/target/releases', 'elasticsearch-%s.tar.gz' % (release)))
+  artifacts.append(os.path.join('distribution/deb/target/releases', 'elasticsearch-%s.deb' % (release)))
+  artifacts.append(os.path.join('distribution/rpm/target/releases', 'elasticsearch-%s.rpm' % (release)))
   return artifacts
 
 def get_artifacts(release):
-  common_artifacts = artifact_names(release, 'target/releases/')
+  common_artifacts = artifact_names(release)
   for f in common_artifacts:
     if not os.path.isfile(f):
       raise RuntimeError('Could not find required artifact at %s' % f)
   return common_artifacts
 
-# Checks the jar files in each package
-# Barfs if any of the package jar files differ
-def check_artifacts_for_same_jars(artifacts):
-  jars = []
-  for file in artifacts:
-    if file.endswith('.zip'):
-      jars.append(subprocess.check_output("unzip -l %s  | grep '\.jar$' | awk -F '/' '{ print $NF }' | sort" % file, shell=True))
-    if file.endswith('.tar.gz'):
-      jars.append(subprocess.check_output("tar tzvf %s  | grep '\.jar$' | awk -F '/' '{ print $NF }' | sort" % file, shell=True))
-    if file.endswith('.rpm'):
-      jars.append(subprocess.check_output("rpm -pqli %s | grep '\.jar$' | awk -F '/' '{ print $NF }' | sort" % file, shell=True))
-    if file.endswith('.deb'):
-      jars.append(subprocess.check_output("dpkg -c %s   | grep '\.jar$' | awk -F '/' '{ print $NF }' | sort" % file, shell=True))
-  if len(set(jars)) != 1:
-    raise RuntimeError('JAR contents of packages are not the same, please check the package contents. Use [unzip -l], [tar tzvf], [dpkg -c], [rpm -pqli] to inspect')
-
-# Generates sha1 checsums for all files
-# and returns the checksum files as well
-# as the given files in a list
-def generate_checksums(files):
-  res = []
-  for release_file in files:
-    directory = os.path.dirname(release_file)
-    file = os.path.basename(release_file)
-    checksum_file = '%s.sha1.txt' % file
-
-    if os.system('cd %s; shasum %s > %s' % (directory, file, checksum_file)):
-      raise RuntimeError('Failed to generate checksum for file %s' % release_file)
-    res = res + [os.path.join(directory, checksum_file), release_file]
-  return res
-
-def download_and_verify(release, files, plugins=None, base_url='https://download.elastic.co/elasticsearch/elasticsearch'):
+# Sample URL:
+# http://download.elasticsearch.org/elasticsearch/release/org/elasticsearch/distribution/elasticsearch-rpm/2.0.0-beta1-SNAPSHOT/elasticsearch-rpm-2.0.0-beta1-SNAPSHOT.rpm
+def download_and_verify(release, files, plugins=None, base_url='https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution'):
```

Comment:
Just a note here. We decided that by convention we will use the same naming as maven.
`groupId` has now changed to `org.elasticsearch.distribution.[packaging]` so I think we should also reflect that change here and use `org/elasticsearch/distribution/[packaging]` where `packaging` is:

* rpm
* deb
* zip
* tar